### ISSUE TITLE
tsfresh 0.20.0: TypeError exception with Pandas 2.0.0

### ISSUE THREAD

**Issue by jpoliv** [blue-yonder/tsfresh#1013]
**The problem**:

A TypeError exception is thrown when executing the example code of the [Quick Start](https://tsfresh.readthedocs.io/en/latest/text/quick_start.html) documentation page.

```
...
>>> extracted_features = extract_features(timeseries, column_id="id", column_sort="time")
Feature Extraction: 100%|████████████████████████████████████████| 15/15 [00:07<00:00,  1.93it/s]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/joliveira/sandbox/py311env/lib64/python3.11/site-packages/tsfresh/feature_extraction/extraction.py", line 164, in extract_features
    result = _do_extraction(
             ^^^^^^^^^^^^^^^
  File "/home/joliveira/sandbox/py311env/lib64/python3.11/site-packages/tsfresh/feature_extraction/extraction.py", line 304, in _do_extraction
    return_df = data.pivot(result)
                ^^^^^^^^^^^^^^^^^^
  File "/home/joliveira/sandbox/py311env/lib64/python3.11/site-packages/tsfresh/feature_extraction/data.py", line 94, in pivot
    return_df.index = return_df.index.astype(self.df_id_type)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joliveira/sandbox/py311env/lib64/python3.11/site-packages/pandas/core/indexes/multi.py", line 3637, in astype
    raise TypeError(
TypeError: Setting a MultiIndex dtype to anything other than object is not supported
>>>
```

**Anything else we need to know?**:

Workaround:
* downgrading Pandas from version 2.0.0 to version 1.5.3 solved the problem

Python 3.11 specific installation note:
* Install the Release Candidate 1 of numba 0.57 (pip install numba==0.57.0rc1);  numba 0.56 fails to install (at least in Fedora 37)

**Environment**:

- Python version: Python 3.10 and Python 3.11
- Operating System: CentOS Stream 8 and Fedora 37
- tsfresh version: 0.20.0
- Install method (conda, pip, source): pip

**Comment by jpoliv** [comment]
Temporary requirements tweak:
```
diff --git a/requirements.txt b/requirements.txt
index f4eaba7..ac8bf7c 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 requests>=2.9.1
 numpy >=1.15.1, <1.24
-pandas>=0.25.0
+pandas>=0.25.0, <2.0.0
 scipy>=1.2.0
 statsmodels>=0.13
 patsy>=0.4.1
```

**Comment by goto-loop** [comment]
The issue seems to be connected to the id column's dtype. If you convert it to `object`, the feature extraction will run just fine: 
```python
timeseries["id"] = timeseries["id"].astype(object)
```

**Comment by nils-braun** [comment]
Thank you for your patience! I have just released tsfresh version 0.20.1 which fixes this incompatibility with pandas 2.0. 
Please try again and re-open a new issue if you still face issues. Thank you!
